app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/CurrentWorkoutActivity.java

```diff
@@ -207,6 +209,26 @@ protected void onDestroy() {
         mLapCounterService = null;
     }
 
+    private void alertToSaveBeforeRestart(String alertText) {
+        // Ask if they'd like to save their current workout.
+        new android.support.v7.app.AlertDialog.Builder(this)
+                .setMessage(alertText)
+                .setPositiveButton(R.string.yes, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        saveWorkout();
+                        restartWorkout();
+                    }
+                })
+                .setNegativeButton(R.string.no, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        restartWorkout();
```

Comment:
restarting the workout clears the lap count, but the timer is still running for some reason...

Reply:
oh I see what this problem is at least, try calling `pause()` just before these lines about restarting, it's a separate function call!